tests/Domain/Integrations/Controllers/IntegrationControllerTest.php

```diff
@@ -88,11 +207,447 @@ public function test_it_can_active_an_integration_with_an_organization(): void
+            'url' => 'https://localhost:3000',
+        ]);
+
+        $response->assertRedirect("/nl/integraties/{$integration->id}");
+
+        $this->assertDatabaseHas('integrations_urls', [
+            'environment' => Environment::Testing->value,
+            'type' => IntegrationUrlType::Callback->value,
+            'url' => 'https://localhost:3000',
+        ]);
+    }
+
+    public function test_it_cant_store_an_integration_url_if_unauthorized(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
+
+        $integration = $this->givenThereIsAnIntegration();
+
+        $response = $this->post("/integrations/{$integration->id}/urls", [
+            'environment' => Environment::Testing->value,
+            'type' => IntegrationUrlType::Callback->value,
+            'url' => 'https://localhost:3000',
+        ]);
+
+        $response->assertForbidden();
+
+        $this->assertDatabaseMissing('integrations_urls', [
+            'environment' => Environment::Testing->value,
+            'type' => IntegrationUrlType::Callback->value,
+            'url' => 'https://localhost:3000',
+        ]);
+    }
+
+    public function test_it_can_destroy_an_integration_url(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
+
+        $integration = $this->givenThereIsAnIntegration();
+        $this->givenTheActingUserIsAContactOnIntegration($integration);
+        $integrationUrl = $this->givenThereIsALoginUrlForIntegration($integration);
+
+        $response = $this->delete("/integrations/{$integration->id}/urls/{$integrationUrl->id}");
+
+        $response->assertRedirect("/nl/integraties/{$integration->id}");
+
+        $this->assertDatabaseMissing('integrations_urls', [
+            'id' => $integrationUrl->id,
+        ]);
+    }
+
+    public function test_it_cant_destroy_an_integration_url_if_unauthorized(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
+
+        $integration = $this->givenThereIsAnIntegration();
+        $integrationUrl = $this->givenThereIsALoginUrlForIntegration($integration);
+
+        $response = $this->delete("/integrations/{$integration->id}/urls/{$integrationUrl->id}");
+
+        $response->assertForbidden();
+
+        $this->assertDatabaseHas('integrations_urls', [
+            'id' => $integrationUrl->id,
+        ]);
+    }
+
+    public function test_it_can_update_integration_urls(): void
+    {
+        $this->actingAs(UserModel::createSystemUser(), 'web');
+
+        $integration = $this->givenThereIsAnIntegration();
+        $this->givenTheActingUserIsAContactOnIntegration($integration);
+        $urls = $this->givenThereAreMultipleUrlsForIntegration($integration);
+
+        $response = $this->patch("/integrations/{$integration->id}/urls", [
+            'loginUrl' => [
+                'id' => $urls->loginUrl->id->toString(),
+                'url' => 'https://updated.test',
```

Comment:
I would give the URLs different values to make sure the correct mapping is done etc.

Reply:
Done